pkg/diagnostics/grpc_tracing.go

```diff
@@ -67,6 +61,9 @@ func GRPCTraceUnaryServerInterceptor(appID string, spec config.TracingSpec) grpc
 			}
 		}
 
+		traceContextBinary := propagation.Binary(span.SpanContext())
+		grpc.SetHeader(ctx, metadata.Pairs(grpcTraceContextKey, string(traceContextBinary)))
```

Comment:
isn't it overwriting tracing header for invoke-service api?

Reply:
That's right, I had this discussion with @shalabhms , he is already working on a PR for service invocation, it will be handled in that PR. In the meantime can I leave a comment "TODO: Handle the case of service invocation"?

Reply:
@pruthvidhodda ,  we need to check for non- invoke service api . We planned to remove adding trace headers in InvokeService method but I see further that it wont work. So please add the check.

Reply:
Can I safely assume that all the non-invocation api responses will not have this header already set in grpc metadata?

Reply:
Yes, based on current code, your assumption is valid.